spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/config/MongoDbOutboundGatewayParserTests.java

```diff
@@ -109,6 +110,20 @@ public void fullConfigWithMongoTemplate() {
 		assertEquals("foo", TestUtils.getPropertyValue(gateway, "collectionNameExpression.literalValue"));
 	}
 
+	@Test
+	public void fullConfigWithCollectionCallback() {
+		MongoDbOutboundGateway gateway = TestUtils.getPropertyValue(
+				context.getBean("fullConfigWithCollectionCallback"), "handler", MongoDbOutboundGateway.class);
```

Comment:
There is really a bean with the name `fullConfigWithCollectionCallback.handler`: http://docs.spring.io/spring-integration/docs/4.3.6.RELEASE/reference/html/messaging-endpoints-chapter.html#endpoint-inner